src/rust/include/rust/types.h

```diff
@@ -8,9 +8,9 @@
 #include <stdint.h>
 
 #ifdef WIN32
-typedef uint16_t codeunit;
+typedef char16_t codeunit;
 #else
-typedef uint8_t codeunit;
+typedef char codeunit;
```

Comment:
I originally used `char8_t`, but [had to switch to `char`](https://github.com/zcash/zcash/compare/a68516286ba4d7f24086ad70382c14e81cae7fa9..28aa9ad1e0ea35babe880503f398ae95e86e9c0d) because that's the type that `cxx` uses for its [`rust::String` constructor](https://cxx.rs/binding/string.html) on non-Windows platforms.

Reply:
Ugh. Well, casts between `char *` and `unsigned char *`-like types are guaranteed not to lose information on POSIXish platforms, at least. I guess it's probably okay, but please review all uses.

Reply:
The only other use of `codeunit` in the codebase is for the log path passed to `tracing_init`. After this change, we are casting across the FFI from `const char* / const char16_t*` on the C++ side to `*const u8 / *const u16` on the Rust side. This has no functional change, as previously we were doing that cast on the C++ side when casting from `std::string.c_str()`'s output to `codeunit`.